lib/binary_to_decimal.rb

```diff
@@ -1,9 +1,18 @@
-# A method named `binary_to_decimal` that receives as input an array of size 8.
+# A method named `binary_to_decimal` that
+#receives as input an array of size 8.
 # The array is randomly filled with 0’s and 1’s.
 # The most significant bit is at index 0.
 # The least significant bit is at index 7.
-# Calculate  and return the decimal value for this binary number using
+# Calculate  and return the decimal value for
+# this binary number using
 # the algorithm you devised in class.
+
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  index = binary_array.length - 1
+  result = 0
+  while index >= 0
+    result += binary_array[index] * (2 ** (7 - index))
```

Comment:
I suggest making 7 a constant or using the `binary_array.length -1` just to make the code a bit more flexible.  Using specific numbers like 7 in code are sometimes called ["magic numbers"](https://codeburst.io/software-anti-patterns-magic-numbers-7bc484f40544#targetText=A%20magic%20number%20is%20a,so%20they%20should%20be%20avoided.).